contract/contracts/Treasury/EVMTreasury.sol

```diff
@@ -124,22 +126,34 @@ contract EVMTreasury is Pausable, ReentrancyGuard, IEVMTreasury {
     }
 
     /* ========== LIGHTCLIENT FUNCTIONS ========== */
-
-    /// @notice The argument types and logic need to be replaced with the proper types
-    function update_light_client(string memory header, string memory proof) public whenNotPaused {
+    function update_light_client(
+        bytes calldata header,
+        bytes[] calldata proof
+    ) public whenNotPaused {
+        Verify.BlockHeader memory block_header = Verify.parse_header(header);
         require(
-            keccak256(abi.encodePacked(proof)) == keccak256(abi.encodePacked("valid")),
-            "EVMTreasury::update_light_client: Invalid block finalization proof"
+            Verify.verify_header_to_header(client.last_header, header),
+            "EVMTreasury::update_light_client: Invalid header"
+        );
+        require(
+            Verify.verify_finalization_proof(block_header, keccak256(header), proof),
```

Comment:
How about saving the proof in the `Client` storage and skip the re-verification process in the `verify_header_to_header` function?

Reply:
What does the re-verification process mean?

Reply:
The current block proof is re-verified when the next block is updated, in the `verifyHeaderToHeader` function. Is there a chance for the proof to be updated at the next block?

Reply:
Yes, but it seems intended. You can check [here](https://github.com/postech-dao/simperby/blob/main/common/src/light_client.rs)